nustate/persistence/store.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package persistence
+
+import (
+	"github.com/juju/juju/nustate/model"
+	"github.com/juju/juju/nustate/persistence/transaction"
+)
+
+// An abstraction for a model backing store
+type Store interface {
```

Comment:
How do you envisage the granularity here?

- One giant interface with every query.
- One giant interface embedding many model-concern-grouped interfaces with topical queries.
- Many store interfaces with methods grouped by model/operation concern.
- A store per (notwithstanding DRY) operation.

Reply:
I didn't flesh this out intentionally to see what people thought about this. My thought was to break it down into a bunch entity-specific interfaces and aggregate them into the store. This way, we could apply the interface segregation principle when creating operations and/or defining things that interact with the store.

Reply:
See: https://github.com/juju/juju/pull/11871/files#r467761015